torch/_export/__init__.py

```diff
@@ -62,7 +65,157 @@
 from .wrappers import _wrap_submodules
 
 
-def dynamic_dim(t: torch.Tensor, index: int):
+class _Dim(type):
+    """
+    Metaclass for Dim types.
+    """
+    pass
+
+
+def Dim(name, *, min=None, max=None):
+    """
+    Constructs a type analogous to a named symbolic integer with a range.
+    It can be used to describe multiple possible values of a tensor dimension.
+    Note that different dimensions of the same tensor, or of different tensors,
+    can be described by the same type.
+    """
+    _min = 2 if min is None else min
+    _max = sys.maxsize if max is None else max
+    return _Dim(name, (int,), {"min": _min, "max": _max})
+
+
+def dims(*names: str):
```

Comment:
Just making sure I understand this correctly: users are not supposed to modify min/max of created Dim objects, and dims doesn't take max/min arguments, so `dims` can only be used to attach names to dims?

Reply:
Sorry, this is an oversight. I should add min/max args here.